test/jdk/javax/swing/JFileChooser/FileFilterDescription/FileFilterDescription.java

```diff
@@ -21,44 +21,56 @@
  * questions.
  */
 
-import java.applet.Applet;
 import java.io.File;
-
+import java.awt.BorderLayout;
 import javax.swing.JFileChooser;
+import javax.swing.JFrame;
 import javax.swing.SwingUtilities;
 import javax.swing.UIManager;
 import javax.swing.UnsupportedLookAndFeelException;
 import javax.swing.filechooser.FileFilter;
 
-public final class FileFilterDescription extends Applet {
+/*
+ * @test
+ * @bug 8029536
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual FileFilterDescription
+ */
+public final class FileFilterDescription {
 
-    @Override
-    public void init() {
-    }
-
-    @Override
-    public void start() {
-        try {
-            test();
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-    }
+     private static final String INSTRUCTIONS = """
+         1) Check that current filter in the opened JFileChooser is a "CustomFileFilter".
+         2) Close the JFileChooser.
+         3) Test will repeat steps 1 - 2 for all supported look and feels.
+         4) If it's true for all look and feels then click Pass else click Fail.  """;
 
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame passFailJFrame = PassFailJFrame.builder()
+                .title("JFileChooser Filefilter Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows(10)
+                .columns(35)
```

Comment:
I guess testUI and awaitAndCheck can be used here only.

Reply:
It may, but it wouldn't work well. Prasanta's approach works better.